### DISCUSSION TITLE
Renaming the CRD's name to KamajiControlPlaneProvider

### DISCUSSION THREAD

**Discussion by jds9090** [clastix/cluster-api-control-plane-provider-kamaji#145] [Ideas]
The main role of Kamaji Control Plane is just for linking with CAPI. So, we need to check TCP(tenant Control Plane) if we want to check the status of control plane in detail.

That's why [this repo](https://github.com/clastix/cluster-api-control-plane-provider-kamaji)'s name is cluster-api-control-plane-provider-kamaji, not cluster-api-control-plane-kamaji.

I suggest renaming the CRD's name to KamajiControlPlaneProvider.

```
$: kubectl get kamajicontrolplaneprovider
NAME                       VERSION   INITIALIZED   READY   AGE
capi-quickstart   1.23.10   true          true    60d
```

Related to #66

**Comment by prometherion** [comment]
I'm up for a discussion about this, although the renaming would be hard to tackle.

1. You would need to create all the new resources with the new name.
2. After that, all the `TenantControlPlane` instances must be patched with the right `OwnerReference` to the new resource
3. Finally, the old resources can be deleted, as well as the old CRD

I'm keeping the issue open to collect feedback from other users.

**Comment by jds9090** [comment]
> I'm up for a discussion about this, although the renaming would be hard to tackle.
> 
> 1. You would need to create all the new resources with the new name.
> 2. After that, all the `TenantControlPlane` instances must be patched with the right `OwnerReference` to the new resource
> 3. Finally, the old resources can be deleted, as well as the old CRD
> 
> I'm keeping the issue open to collect feedback from other users.

Thank you for your consideration :)
